### PR TITLE
Add FAQ entry for emulators

### LINKED ISSUE #859: Provide documentation on using emulators
We're not baking emulator support into the client libraries at the moment, but we should provide documentation for how to do it.

### PR DESCRIPTION
**Pull request by jskeet**
Fixes #859.

### CODE REVIEW

File: docs/root/faq.md [hunk 1]
```diff
@@ -106,3 +106,32 @@ it yourself to integrate with other systems - see the
...
+
+However, it is quite simple to manually create the appropriate
+client for gRPC-based APIs, by first creating a channel and then
+passing that to the static `Create` method of the relevant client
+class. Note that in current emulator implementations, the channel
+credentials must be set to `ChannelCredentials.Insecure`.
+
+Example for PubSub:
+
+```csharp
```
Comment (chrisdunelm) [inline docs/root/faq.md]: Can this not be pulled from a snippet?

File: docs/root/faq.md [hunk 2]
```diff
@@ -106,3 +106,32 @@ it yourself to integrate with other systems - see the
...
+string emulatorHostAndPort = Environment.GetEnvironmentVariable("PUBSUB_EMULATOR_HOST");
```
Comment (chrisdunelm) [inline docs/root/faq.md]: Can you show an example of what the contents of the variable "PUBSUB_EMULATOR_HOST" should look like?
("localhost:5000" I'm guessing)

File: docs/root/faq.md (hunk 1 above)
Comment (jskeet) [inline docs/root/faq.md]: Potentially. Will give it a try. We definitely don't want to try executing it, of course.

File: docs/root/faq.md (hunk 2 above)
Comment (jskeet) [inline docs/root/faq.md]: Sure, will add a comment.

**Comment by jskeet** [comment]
(Nits addressed - will merge when green.)

**Pull request merged**